### DISCUSSION TITLE
Delimiter start and end div by h1

### DISCUSSION THREAD

**Discussion by eerison** [markdoc/markdoc#367] [Q&A] [answered]
Is there anyway to delimiter start and end of div by h1 or other tag?

Example 

```markdown
# Title 1
 Content 

# Title 2
Content 
```

```html
<div>
<h1>Title 1</h1>
Content
</div>

<div>
<h1>Title 2</h1>
Content 
</div>
```

**Comment by rpaul-stripe** [comment, marked as answer]
I would recommend adding a new custom tag and explicitly wrapping the sections, but if you didn't want to do that you could potentially write an AST transform that walks the document looking for H1 nodes and then wraps the node and all subsequent sibling nodes up to the next H1 or the end of the enclosing parent.

The following code shows how to find all of the H1 headings in the top-level document node:

```
const doc = Markdoc.parse(example);
for (const child of doc.children) 
  if (child.type === "heading" && child.attributes.level == 1)
    console.log(child);
```

**Reply by eerison** [reply]
🤯 thank you for the tip, I will try that :)

**Reply by eerison** [reply]
Hey @rpaul-stripe 

ok I could get the child that contains H1, But I'm not sure how to put this inside of div/tag
